### ISSUE TITLE
Prevent overfitting

### ISSUE THREAD

**Issue by dev079144** [dev079144/the-old-man-and-the-cpp#2]
- [ ] Learn about and implement dropout
- [ ] Split data into train/validate (10-15%)
- [ ] Weight decay (L2 regularization)
- [ ] Learning rate scheduling
- [ ] Gradient clipping
- [ ] Early stopping
- [ ] Loss monitoring
- [ ] Data augmentation
- [ ] Other ...
